### PR TITLE
wand.js + bubble.js = bubblewand

### PR DESCRIPTION
**Pull request by imgntn**
Attach the wand.js script to an entity and then move it around to make bubbles.  Customize bubble behavior with bubble.js - make sure to update the path reference in wand.js

Still working on sound stability.  Need to standardize some of the function scope (some stuff is on the entity, some is on a singleton, because it didn't start as an entity script.)

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
An admin must verify this patch for the pull request build to fire. Type "ok to test" to begin testing this pull request or "add to whitelist" to test this and all future pull requests from this collaborator.

File: examples/toys/bubblewand/bubble.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+(function() {
```
Comment (ZappoMan) [inline examples/toys/bubblewand/bubble.js]: we need to get a file header with license info added to this file.

File: examples/toys/bubblewand/createWand.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+function randInt(min, max) {
```
Comment (ZappoMan) [inline examples/toys/bubblewand/createWand.js]: file header

File: examples/toys/bubblewand/wand.js [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+function convertRange(value, r1, r2) {
+    return (value - r1[0]) * (r2[1] - r2[0]) / (r1[1] - r1[0]) + r2[0];
+}
+
+// helpers -- @zappoman
+// Computes the penetration between a point and a sphere (centered at the origin)
+// if point is inside sphere: returns true and stores the result in 'penetration'
+// (the vector that would move the point outside the sphere)
+// otherwise returns false
+function findSphereHit(point, sphereRadius) {
```
Comment (ZappoMan) [inline examples/toys/bubblewand/wand.js]: these functions now exist in utils.js

File: examples/toys/bubblewand/wand.js [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+                        dimensions.z += 0.015 * convertedVolume;
+
+                    } else {
+                        dimensions.x += 0.015 * velocityStrength;
+                        dimensions.y += 0.015 * velocityStrength;
+                        dimensions.z += 0.015 * velocityStrength;
+                    }
+
+                }
+            } else {
```
Comment (ZappoMan) [inline examples/toys/bubblewand/wand.js]: what's going on here? double else??

File: examples/toys/bubblewand/wand.js (hunk 3 above)
Comment (jherico) [inline examples/toys/bubblewand/wand.js]: we should create findSphereLineSegementHit() and findSphereRayHit() methods as well

File: examples/toys/bubblewand/wand.js (hunk 4 above)
Comment (imgntn) [inline examples/toys/bubblewand/wand.js]: not sure, some kind of naughty pasting error i'd guess!

**Pull request merged**